.changeset/fair-seahorses-arrive.md

```diff
@@ -0,0 +1,10 @@
+---
+"create-cloudflare": minor
```

Comment:
Should this be minor? Changing a minimum node version is usually considered major

Reply:
Technically, yes, it should be major. However, we've talked about this a good deal internally and we decided to make it a minor bump for a couple of reasons. 

Changing a minimum node version is usually considered major, but most of the time this is done for a different reason. Usually it means that the module in question is using features that are only available in newer version of node, and running them on older versions won't work anymore.

That's not really the case here. What's happening is that our upstream dependencies are themselves dropping support for node and changing their engines spec, all at different cadences. So the answer to `Does c3 work with node v16` is: Sometimes, it depends on the framework you're using. 

In our case we'd like to change our engines spec because we started testing exclusively on v18, and it feels a little dis-ingenuous to claim we support v16 support since we know it will break in some cases, and we don't be detecting new failures as they arise with different `create-*` tools.

If we were to strictly follow semver, it seems like we'd need to bump a major version each time we bump to a new major version of a framework, since technically it's a breaking change if that's the framework you use. This leaves us with 2 options: do major bumps as frequently as we bump our dependencies, or keep stale dependencies around longer so we can batch them up into major bumps of c3. Neither seem like great options.

There's also the unfortunate reality that frameworks don't often align the major versions of their framework in respective scaffolding tools. Sometimes a new minor version of `create-*` will use a new major version of `framework-*`. 

Major bumps are a little challenging for us logistically. We've got calls to `create cloudflare@2` throughout `wrangler` itself, and in tons of documentation that would be tough to update in a coordinated manner. 

Another issue is that C3 now automatically run the latest version if it detects one is available, but only if it's a minor version difference. We do this since most people use `npx` or it's counterpart in other package managers, which cache versions of these pretty aggressively. 

This means that as we release major versions of c3, many users would get stuck on older versions, which we don't really want at the moment since many frameworks are churning to drop node v16 support.

Reply:
I agree with these justifications for keeping this a minor bump, not major.

Just want to call out that wrangler's calls to `create cloudflare@2` are to protect old installations of wrangler. When C3 needs a major bump, wrangler will update those calls to the latest major for future releases

Reply:
Appreciate the explanation - no complaints there 🙂